### PR TITLE
Move likes

### PR DESCRIPTION
**Pull request by JakeDawkins**
## Changes

- Added filter in the userFeed resolver for likes
- Added likes model
- Added likes resolver 
- Added mutation for likes

## Tests

- likes model
- likes resolver
- likes filter in userFeed

### CODE REVIEW

File: src/rock/models/likes/resolver.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+// import { flatten } from "lodash";
+
+export default {
+  Mutation: {
+    toggleLike(_, { nodeId }, { models, user }) {
+      // XXX what should the response be if invalid/insufficient data?
```
Comment (jbaxleyiii) [inline src/rock/models/likes/resolver.js:11]: @JakeDawkins we have a generic mutation response that we should implement here. https://github.com/NewSpring/Heighliner/blob/30577753b3805a1dc31dd07ef28383cfdfe94d9d/src/util/node/schema.js#L7-L11

File: src/rock/models/likes/mutations.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+export default [
+   "toggleLike(nodeId: String!): [Node]",
```
Comment (jbaxleyiii) [inline src/rock/models/likes/mutations.js]: This should be a `MutationResponse` that adds a `like: Node` which isn't an array. That way the client can easily update its client cache

**Review by jbaxleyiii — CHANGES_REQUESTED** [review]

**Comment by JakeDawkins** [comment]
@jbaxleyiii whenever I try to start heighliner with these changes, I get `Error: Type "LikesMutationResponse" not found in document.` The schema is there an imported. I haven't implemented it yet in the resolver, but any ideas?

**Comment by jbaxleyiii** [comment]
@JakeDawkins hmm, it should be erroring for missing resolver instead. Can you console log the schema files to make sure it is generating correctly? And maybe add the resolver for it?

File: src/rock/models/likes/model.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ export class Like {
   }
 
   async toggleLike(nodeId, userId, nodeModel) {
-    const existingLike = await this.model.findOne({
```
Comment (jbaxleyiii) [inline src/rock/models/likes/model.js]: This should still be able to be const yes?

File: src/rock/models/likes/model.js [hunk 4]
```diff
@@ -54,9 +54,16 @@ export class Like {
         createdAt: new Date(),
       });
     }
+
     const guid = createGlobalId(userId);
     await this.cache.del(guid);
-    return this.getLikedContent(userId, nodeModel);
+
+    return ({
+      like: nodeModel.get(nodeId),
```
Comment (jbaxleyiii) [inline src/rock/models/likes/model.js:62]: 💯

**Review by jbaxleyiii — APPROVED** [review]

**Pull request merged**